clarity/src/vm/types/signatures.rs

```diff
@@ -74,12 +74,36 @@ pub struct TupleTypeSignature {
     type_map: BTreeMap<ClarityName, TypeSignature>,
 }
 
+#[cfg(feature = "arbitrary")]
+impl arbitrary::Arbitrary<'_> for TupleTypeSignature {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'_>) -> arbitrary::Result<Self> {
+        let type_map = BTreeMap::arbitrary(u)?;
+        Self::try_from(type_map).map_err(|_| arbitrary::Error::IncorrectFormat)
+    }
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
 pub struct BufferLength(u32);
 
+#[cfg(feature = "arbitrary")]
+impl arbitrary::Arbitrary<'_> for BufferLength {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'_>) -> arbitrary::Result<Self> {
+        let len = u32::arbitrary(u)?;
```

Comment:
In order to decrease the number of obviously invalid tests, would it make sense to bound the `len` here to be something smaller than 2**32 - 1?

Reply:
The `try_from` on the next line will report an error if it is out of range, causing the fuzzer to pick a new number.